gitprovider/types_repository.go

```diff
@@ -189,12 +189,16 @@ type CommitInfo struct {
 	TreeSha string `json:"tree_sha"`
 }
 
-// CommitFile contains high-level information about a file added to a commit.
-type CommitFile struct {
+// File contains high-level information about a file either add it to a commit or for content downloads
+type File struct {
 	// Path is path where this file is located.
 	// +required
 	Path *string `json:"path"`
 
+	// Name is path where this file is located.
+	//
+	Name *string `json:"name"`
+
```

Comment:
Could you do this by adding a new struct rather than renaming the existing one and adding a field which is sometimes redundant?